docusaurus/docs/how-to-guides/data-source-plugins/profiling-backend-plugin.md

```diff
@@ -0,0 +1,121 @@
+---
+id: profiling-backend-plugin
+title: Profiling a backend plugin
+description: How to profile a backend plugin.
+keywords:
+  - grafana
+  - plugins
+  - plugin
+  - profile
+  - profiling
+  - continuos profiling
+  - backend
+  - back-end
+---
+
+You can configure a backend plugin to enable certain diagnostics when it starts, so-called profiling data. This can be useful
+when investigating certain performance problems, such as high CPU or memory usage, or enabling [continous profiling](https://grafana.com/oss/pyroscope/).
+
+## Configure profiling
+
+In the [Grafana configuration file](https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/) you can configure profiling under a `[plugin.<plugin ID>]` section where `<plugin ID>` is the plugin identifier of your backend plugin you want to profile, e.g. [grafana-github-datasource](https://grafana.com/grafana/plugins/grafana-github-datasource/).
+
+Running a backend plugin with profiling enabled and without block and mutex profiling enabled should only add a fraction of overhead and is suitable for production or continuous profiling scenarios. Adding a small fraction of block and mutex profiling, such as 10-5 (10%-20%) should in general be fine, but each plugin might vary.
```

Comment:
People (like myself) may not know what block and mutex mean. You probably need to do an intro to what these parameters mean and point to some documentation for more information.

Reply:
Ah, I see you are explaining this below, then a link to those sections should suffice.

Reply:
Tried to improve this moving it to its own section. WDYT?

Reply:
Added a pointer that options are detailed in sub-sections. WDYT?